lib/runtime/wasmer/exports_test.go

```diff
@@ -1210,3 +1210,138 @@ func TestInstance_TransactionPaymentCallApi_QueryCallFeeDetails(t *testing.T) {
 		require.Equal(t, test.expect, details)
 	}
 }
+
+func TestInstance_GrandpaGenerateKeyOwnershipProofEncoding(t *testing.T) {
+	t.Parallel()
+	keyOwnershipProof := types.OpaqueKeyOwnershipProof([]byte{64, 138, 252, 29, 127, 102, 189, 129, 207, 47, 157,
+		60, 17, 138, 194, 121, 139, 92, 176, 175, 224, 16, 185, 93, 175, 251, 224, 81, 209, 61, 0, 71})
+	encoded := scale.MustMarshal(keyOwnershipProof)
+	var proof types.OpaqueKeyOwnershipProof
+	err := scale.Unmarshal(encoded, &proof)
+	require.NoError(t, err)
+	require.Equal(t, keyOwnershipProof, proof)
+}
+
+func TestInstance_GrandpaGenerateKeyOwnershipProof(t *testing.T) {
+	t.Parallel()
+	rt := NewTestInstance(t, runtime.WESTEND_RUNTIME_v0929)
+	identity := common.MustHexToBytes("0x88dc3417d5058ec4b4503e0c12ea1a0a89be200fe98922423d4334014fa6b0ee")
+	identityPubKey, _ := ed25519.NewPublicKey(identity)
+	var authorityID [32]byte
+	copy(authorityID[:], identityPubKey.Encode())
+
+	_, err := rt.GrandpaGenerateKeyOwnershipProof(uint64(0), authorityID)
```

Comment:
Can we check the returned value as well?

Reply:
yeah I can add an empty check for this case (the more integration-y tests are out of scope of this PR)